packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -573,4 +601,13 @@ export class MultipartUserData extends UserData {
       throw new Error(MultipartUserData.USE_PART_ERROR);
     }
   }
+
+  public persist( value: boolean ): void {
+    if (this.defaultUserData) {
+      this.defaultUserData.persist(value);
```

Comment:
In this context, does this need any other check to ensure this is for Windows?

Reply:
I don't think so. It's a pass-though, so if the default userdata is Windows then it will work. If it's Linux, then you'll get the exception thrown from the `LinuxUserData` implementation.